src/Microsoft.OData.Client/Materialization/EntryValueMaterializationPolicy.cs

```diff
@@ -591,7 +591,7 @@ private void MaterializeResolvedEntry(MaterializerEntry entry, bool includeLinks
 
                 foreach (ODataNestedResourceInfo link in entry.NestedResourceInfos)
                 {
-                    MaterializerNavigationLink linkState = MaterializerNavigationLink.GetLink(link, this.MaterializerContext);
+                    MaterializerNestedEntry linkState = MaterializerNestedEntry.GetNestedEntry(link, this.MaterializerContext);

```

Comment:
Can we get rid of the `MaterializerNavigationLink`? since we've replaced it with the `MaterializerNestedEntry`? Will it require a lot of refactoring?

Reply:
Fixed.